Tests/MongoSwiftSyncTests/UnifiedTestRunner/UnifiedOperation.swift

```diff
@@ -0,0 +1,129 @@
+import Foundation
+import MongoSwiftSync
+
+struct UnifiedOperation: Decodable {
+    /// Represents an object on which to perform an operation.
+    enum Object: RawRepresentable, Decodable {
```

Comment:
I made some somewhat arbitrary feeling choices about which types to nest here or not, e.g. `Object` is nested but `UnifiedOperationResult` is not, `ExpectedError` could arguably be nested in `UnifiedOperationResult` but isn't, etc.
Ultimately it came down to how complex a type "felt" to me; `Object` is quite simple but the other two are less so. 

I don't know if we've ever really standardized on when to do this in test code though, let me know if you have opinions.

Reply:
I've just been doing it by feel also. Or if a type is only used in the context of another type.